libraries/networking/src/NLPacket.cpp

```
@@ -0,0 +1,61 @@
+//
+//  NLPacket.cpp
+//  libraries/networking/src
+//
+//  Created by Clement on 7/6/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "NLPacket.h"
+
+int64_t NLPacket::localHeaderSize(PacketType::Value type) {
+    int64_t size = ((NON_SOURCED_PACKETS.contains(type)) ? 0 : NUM_BYTES_RFC4122_UUID) +
+                    ((NON_VERIFIED_PACKETS.contains(type)) ? 0 : NUM_BYTES_RFC4122_UUID);
+    return size;
+}
+
+int64_t NLPacket::maxPayloadSize(PacketType::Value type) {
+    return Packet::maxPayloadSize(type) - localHeaderSize(type);
+}
+
+qint64 NLPacket::totalHeadersSize() const {
+    return localHeaderSize() + Packet::localHeaderSize();
+}
+
+qint64 NLPacket::localHeaderSize() const {
+    return localHeaderSize(_type);
+}
+
+std::unique_ptr<NLPacket> NLPacket::create(PacketType::Value type, int64_t size) {
+    if (size > maxPayloadSize(type)) {
+        return std::unique_ptr<NLPacket>();
```

Should we just fail hard here and assert?